WooCommerce/Classes/ViewRelated/Blaze/TargetLanguages/BlazeTargetLanguagePickerViewModel.swift

```diff
@@ -107,20 +107,20 @@ private extension BlazeTargetLanguagePickerViewModel {
     /// and display the first fetch result immediately.
     ///
     func configureDisplayedData() {
-        $languages.combineLatest($isSyncingData, $syncError)
-            .map { languages, isSyncing, error -> SyncState in
-                if error != nil, languages.isEmpty {
+        $fetchedLanguages.combineLatest($displayedLanguages, $isSyncingData, $syncError)
+            .map { fetchedLanguages, displayedLanguages, isSyncing, error -> SyncState in
+                if error != nil, fetchedLanguages.isEmpty {
                     return .error
-                } else if isSyncing, languages.isEmpty {
+                } else if isSyncing, fetchedLanguages.isEmpty {
                     return .syncing
                 }
-                return .result(items: languages)
+                return .result(items: displayedLanguages)
             }
```

Comment:
Previously, I checked for the displayed items before showing the error or loading state. This would not be correct in search mode, so this change is for checking fetched items instead. The logic is similar for target topics.